tests/test-brillhart.C

```diff
@@ -0,0 +1,87 @@
+// =================================================================== //
+// Copyright(c)'1994-2009 by The Givaro group
+// This file is part of Givaro.
+// Givaro is governed by the CeCILL-B license under French law
+// and abiding by the rules of distribution of free software.
+// see the COPYRIGHT file for more details.
+// Time-stamp: <29 Jun 19 08:40:29 Jean-Guillaume.Dumas@imag.fr>
+// Givaro : Modular square roots
+// =================================================================== //
+
+/*! @file tests/test-modsqroot.C
+ * @ingroup examples
+ * @ingroup integers
+ * @see examples/Integer/ModularSquareRoot.C
+ * @brief NO DOC
+ */
+#include <iostream>
+#include <stdlib.h>
+#include <givaro/givintsqrootmod.h>
+#include <givaro/givtimer.h>
+
+using namespace Givaro;
+IntSqrtModDom<> ISM;
+
+bool TestBrillart(const Integer& p) {
+    Integer a,b;
+    ISM.Brillhart(a,b,p);
+    return ISM.areEqual(a*a+b*b,p);
+}
+
+bool TestSoSmP(const Integer& k, const Integer& p) {
+    Integer a,b;
+    ISM.sumofsquaresmodprime(a,b,k,p);
+    return ISM.isZero( (a*a+b*b-k) % p );
+}
+
+
+
+int main(int argc, char** argv) {
+    int nbtests = (argc>1?atoi(argv[1]):1000);
+    int sizes = (argc>2?atoi(argv[2]):100);
+    unsigned long seed = (unsigned long)(argc>3?(unsigned long)atoi(argv[3]):(unsigned long)BaseTimer::seed ());
```

Comment:
Why use `unsigned long` here while seed are everywhere `int64_t`. We should avoid as much as possible to rely on variable size int types.

Reply:
ok